### ISSUE TITLE
Rustanalyzer in VsCode not indexing

### ISSUE THREAD

**Issue by Roba1993** [rune-rs/rune#284]
While developing I encounter the problem, that the VsCode indexing process is running endless for rune.
My environment is:
stable-x86_64-pc-windows-msvc (default)
rustc 1.55.0 (c8dfcfe04 2021-09-06)

![grafik](https://user-images.githubusercontent.com/1376958/136692376-d89e1712-0ef8-4e4e-a893-a04474bcd981.png)

Has someone else also encountered this issue?

**Comment by udoprog** [comment]
I'm seeing this issue as well! Might be worth [reporting to rust-analyzer](https://github.com/rust-analyzer/rust-analyzer/issues/new/choose).

This happens now when working on rune, so it can be reproduced by checking out this repo.

**Comment by udoprog** [comment]
@lnicola That is really cool. I have two questions for you:

Would you want to do a PR in the interim and how did you figure this out?

**Comment by lnicola** [comment]
Sure, filed #285.

> Edit: Ah, I see you're part of rust-analyzer/rust-analyzer#10544 - Nice job!

`rust-analyzer` handles broken code relatively well, so I removed crates (excluding from `Cargo.toml` doesn't work, probably due to some `path` dependencies) until I pinned it down to the `rune` crate, then I removed one module at a time. I already suspected it was related to macros because of the stack trace in `rust-analyzer/rust-analyzer#10544`.
